cudamapper/src/main.cu

```diff
@@ -225,34 +232,31 @@ int main(int argc, char* argv[])
     };
 
     auto compute_overlaps = [&](const query_target_range query_target_range, const int device_id) {
-        std::vector<std::shared_ptr<std::future<void>>> print_pafs_futures;
-
         cudaSetDevice(device_id);
 
         auto query_start_index = query_target_range.query_range.first;
-        auto query_end_index   = query_target_range.query_range.second;
+        auto query_end_index   = query_target_range.query_range.second - 1;
 
-        std::cerr << "Procecssing query range: (" << query_start_index << " - " << query_end_index - 1 << ")" << std::endl;
+        std::cerr << "Processing query range: (" << query_start_index << " - " << query_end_index << ")" << std::endl;
 
         std::shared_ptr<claragenomics::cudamapper::Index> query_index(nullptr);
         std::shared_ptr<claragenomics::cudamapper::Index> target_index(nullptr);
         std::unique_ptr<claragenomics::cudamapper::Matcher> matcher(nullptr);
 
         {
             CGA_NVTX_RANGE(profiler, "generate_query_index");
-            query_index = get_index(*query_parser, query_start_index, query_end_index, k, w, device_id, all_to_all);
+            query_index = get_index(*query_parser, query_start_index, query_end_index, k, w, device_id, all_to_all, filtering_parameter);
```

Comment:
So `get_index` expects last and not past-the-last index. We should change that for consistency (sorry for not spotting this before)